synapse/handlers/search.py

```diff
@@ -35,8 +35,10 @@ class SearchHandler(BaseHandler):
     def __init__(self, hs):
         super(SearchHandler, self).__init__(hs)
         self._event_serializer = hs.get_event_client_serializer()
+        self.hs = hs
```

Comment:
this looks redundant?

Reply:
`self.hs` is used elsewhere, like `self.hs.get_event_sources()`.

Reply:
I'm not seeing that line in the diff?

Reply:
...and I guess that code didn't work before.

Reply:
ohh I see

SearchHandler inherits from BaseHandler. BaseHandler is a thing that exists only to cause this sort of confusion.